ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/ForeignKeyJoinParamsFactory.java

```diff
@@ -27,18 +29,28 @@ private ForeignKeyJoinParamsFactory() {
   }
 
   public static <KRightT> ForeignKeyJoinParams<KRightT> create(
-      final ColumnName leftJoinColumnName,
+      final Optional<ColumnName> leftJoinColumnName,
+      final Optional<ExpressionEvaluator> expressionEvaluator,
       final LogicalSchema leftSchema,
-      final LogicalSchema rightSchema
+      final LogicalSchema rightSchema,
+      final ProcessingLogger processingLogger
   ) {
     if (rightSchema.key().size() != 1) {
       throw new IllegalStateException("rightSchema must have single column key");
     }
-    return new ForeignKeyJoinParams<>(
-        createKeyExtractor(leftSchema, leftJoinColumnName),
-        new KsqlValueJoiner(leftSchema.value().size(), rightSchema.value().size(), 0),
-        createSchema(leftSchema, rightSchema)
-    );
+    if (leftJoinColumnName.isPresent()) {
+      return new ForeignKeyJoinParams<>(
+          createKeyExtractor(leftSchema, leftJoinColumnName.get()),
```

Comment:
nit: I think it's slightly more readable to create the foreign key extractor in an if-statement and then have a single return statement for building the ForeignKeyJoinParams. Also less code duplication.